ibis_substrait/tests/compiler/test_parity.py

```diff
@@ -176,7 +210,7 @@ def test_filter_groupby():
     )
 
     compiler = SubstraitCompiler()
-    run_parity_tests(grouped_table, datasets, compiler=compiler)
+    run_parity_tests(grouped_table, datasets, compiler=compiler, engines=["acero"])
```

Comment:
Now that there are multiple engines running, we might want to be more deliberate in tracking what's working and what isn't.  I'm reading this line as implicitly saying that "datafusion doesn't pass this test" -- if that's the case, maybe we're better off parametrizing each of these tests by `engine` and then marking the individual failing cases?

Reply:
yes, that's right. I'll probably spin off engines as fixtures and parameterize + mark failing cases.

Reply:
Looks like I'll have to do some refactoring to do that, let's hold this for a little while until I do that. I'll spin off engines as separate components. btw, Ideally I was thinking of relying on work already done in [spark-substrait-gateway](https://github.com/voltrondata/spark-substrait-gateway/tree/main/src/backends). It's more polished and handles some engine-specific corner cases as well.

Reply:
Made the changes, had to get a little creative around markings. please take a look and let me know what you think.